extension/src/popup/basics/layout/View/index.tsx

```diff
@@ -0,0 +1,257 @@
+  children,
+  alignment,
+  contentFooter,
+  ...props
+}: ViewContentProps) => {
+  const { isAppLayout } = useContext(ViewContext);
+
+  return (
+    <div className="View__content" {...props}>
+      <ViewInset alignment={alignment} hasVerticalBorder={isAppLayout}>
+        {children}
+      </ViewInset>
+      {contentFooter ? (
+        <ViewInset
+          alignment={alignment}
+          hasVerticalBorder={isAppLayout}
+          additionalClassName="View__inset__footer"
+        >
+          {contentFooter}
+        </ViewInset>
+      ) : null}
+    </div>
+  );
+};
+
+// Footer
+interface ViewFooterProps {
+  children: React.ReactNode;
+  customHeight?: string;
+  customGap?: string;
+  hasExtraPaddingBottom?: boolean;
+  hasTopBorder?: boolean;
+  isInline?: boolean;
+}
+
+const ViewFooter: React.FC<ViewFooterProps> = ({
+  children,
+  customHeight,
+  customGap,
+  hasExtraPaddingBottom,
+  hasTopBorder,
+  isInline,
+  ...props
+}: ViewFooterProps) => {
+  const customStyle = {
+    ...(customHeight ? { "--View-footer-height": customHeight } : {}),
+    ...(hasExtraPaddingBottom
+      ? { "--View-footer-padding-bottom": "1.5rem" }
+      : {}),
+    ...(customGap ? { "--View-footer-gap": customGap } : {}),
+  } as React.CSSProperties;
+
+  return (
+    <footer className="View__footer" style={customStyle} {...props}>
+      <ViewInset
+        hasVerticalBorder
+        hasTopBorder={hasTopBorder}
+        isInline={isInline}
+      >
+        {children}
+      </ViewInset>
+    </footer>
+  );
+};
+
+// Inset
+interface ViewInsetProps {
+  children: React.ReactNode;
+  // Using wide layout for onboarding and similar views
+  isWide?: boolean;
+  // Align items inline
+  isInline?: boolean;
+  // TODO: handle other cases: "start", "end"
+  alignment?: "center";
+  hasVerticalBorder?: boolean;
+  hasTopBorder?: boolean;
+  additionalClassName?: string;
+}
+
+export const ViewInset: React.FC<ViewInsetProps> = ({
+  children,
+  isWide,
+  isInline,
+  alignment,
+  hasVerticalBorder,
+  hasTopBorder,
+  additionalClassName,
+  ...props
+}: ViewInsetProps) => (
+  <div
+    className={`View__inset ${addStyleClasses([
+      isWide ? "View__inset--wide" : "",
+      isInline ? "View__inset--inline" : "",
+      alignment === "center" ? "View__inset--align-center" : "",
+      hasVerticalBorder ? "View__inset--vertical-border" : "",
+      hasTopBorder ? "View__inset--top-border" : "",
+    ])}${additionalClassName ? ` ${additionalClassName}` : ""}`}
+    {...props}
+  >
+    {children}
+  </div>
+);
+
+// View
+interface ViewComponent {
+  Header: React.FC<ViewHeaderProps>;
+  AppHeader: React.FC<ViewAppHeaderProps>;
+  Content: React.FC<ViewContentProps>;
+  Footer: React.FC<ViewFooterProps>;
+  Inset: React.FC<ViewInsetProps>;
+}
+
+interface ViewLayoutProps {
+  children: React.ReactNode;
+  isAppLayout?: boolean;
+}
+
+export const View: React.FC<ViewLayoutProps> & ViewComponent = ({
```

Comment:
`View` is the component we should use on every screen/page that is its own view. It shouldn't matter if it's a main view or subview. Only overlays and modals are not part of this.